### ISSUE TITLE
Backward compatibilty testing e2e test

### ISSUE THREAD

**Issue by Racer159** [zarf-dev/zarf#1409]
### Describe what should be investigated or refactored
We should create a test that allows us to better guarantee the backwards compatibility with older packages.  This would likely take the form of manually making a package from a version of Zarf we want to be backward compatible to (ensuring all package features were used), uploading it to OCI and then creating a simple deployment test of that package for future Zarf PRs.

### Links to any relevant code
https://github.com/defenseunicorns/zarf/tree/main/src/test/e2e

### Additional context
We are working on ways to be more cross-version compatible and this would give us a lot more assurance that we aren't breaking anything when we make changes.

**Comment by Racer159** [comment]
This is partially implemented here: https://github.com/defenseunicorns/zarf/blob/main/src/test/upgrade-test/previously_built_test.go

**Comment by schristoff-du** [comment]
Instead of creating a compat. matrix to ensure no breaking changes, we will be focusing on creating versioning for Zarf schemas to provide more clarity about compatibility.

**Comment by satscottfoster** [comment]
I agree with @schristoff-du because this sounds like a lot of work to satisfy situations which can easily be avoided. Full(ish) backwards compatibility seems like a lot of time create a "nice to have" feature.

Perhaps I am not understand this correctly? Seems to me you can use an older version of Zarf to work with a package created with an older version of Zarf?

**Comment by brandtkeller** [comment]
Given that we have some compatibility testing to date and intent to iterate on versioned schemas - Going to close this as complete. Please reopen if you feel strongly otherwise.
